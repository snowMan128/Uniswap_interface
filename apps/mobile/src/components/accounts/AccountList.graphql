query AccountList(
  $addresses: [String!]!
  $valueModifiers: [PortfolioValueModifier!]
) {
  portfolios(
    ownerAddresses: $addresses
    chains: [ETHEREUM, POLYGON, ARBITRUM, OPTIMISM, BASE, BNB]
    valueModifiers: $valueModifiers
  ) {
    id
    ownerAddress
    tokensTotalDenominatedValue {
      value
    }
  }
}
