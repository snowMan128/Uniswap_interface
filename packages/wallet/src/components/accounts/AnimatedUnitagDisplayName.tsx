import { impactAsync } from 'expo-haptics'
import { useState } from 'react'
import { LayoutChangeEvent } from 'react-native'
import { AnimatePresence, Flex, Icons, Text, TouchableArea } from 'ui/src'
import { IconSizeTokens } from 'ui/src/theme'
import { pushNotification } from 'wallet/src/features/notifications/slice'
import { AppNotificationType, CopyNotificationType } from 'wallet/src/features/notifications/types'
import { UNITAG_SUFFIX } from 'wallet/src/features/unitags/constants'
import { DisplayName, DisplayNameType } from 'wallet/src/features/wallet/types'
import { useAppDispatch } from 'wallet/src/state'
import { sanitizeAddressText, shortenAddress } from 'wallet/src/utils/addresses'
import { setClipboard } from 'wallet/src/utils/clipboard'

type AnimatedUnitagDisplayNameProps = {
  displayName: DisplayName
  unitagIconSize?: IconSizeTokens | number
  address?: string
}

export function AnimatedUnitagDisplayName({
  displayName,
  unitagIconSize = '$icon.24',
  address,
}: AnimatedUnitagDisplayNameProps): JSX.Element {
  const dispatch = useAppDispatch()
  const [showUnitagSuffix, setShowUnitagSuffix] = useState(false)
  const [textWidth, setTextWidth] = useState(0)
  const isUnitag = displayName?.type === DisplayNameType.Unitag

  const onTextLayout = (event: LayoutChangeEvent): void => {
    setTextWidth(event.nativeEvent.layout.width)
  }

  const onPressUnitag = (): void => {
    setShowUnitagSuffix(!showUnitagSuffix)
  }

  const onPressCopyAddress = async (): Promise<void> => {
    if (!address) {
      return
    }

    await impactAsync()
    await setClipboard(address)
    dispatch(
      pushNotification({
        type: AppNotificationType.Copied,
        copyType: CopyNotificationType.Address,
      })
    )
  }

  return (
    <Flex onPress={isUnitag ? onPressUnitag : undefined}>
      <Flex row>
        <Text color="$neutral1" variant="subheading1">
          {displayName.name}
        </Text>
        <AnimatePresence>
          <Flex row animation="semiBouncy" ml={-textWidth} x={showUnitagSuffix ? textWidth : 0}>
            <Flex onLayout={onTextLayout}>
              <Text
                animation="semiBouncy"
                color="$neutral3"
                opacity={showUnitagSuffix ? 1 : 0}
                variant="subheading1">
                {UNITAG_SUFFIX}
              </Text>
            </Flex>
            {isUnitag ? (
              <Flex animation="semiBouncy" pl="$spacing4">
                <Icons.Unitag size={unitagIconSize} />
              </Flex>
            ) : null}
            {address && (
              <TouchableArea
                hapticFeedback
                hitSlop={20}
                pl="$spacing8"
                onPress={onPressCopyAddress}>
                <Flex row alignItems="center" gap="$spacing4">
                  <Text color="$neutral3" numberOfLines={1} variant="body2">
                    {sanitizeAddressText(shortenAddress(address))}
                  </Text>
                  <Icons.CopyAlt color="$neutral3" size="$icon.16" />
                </Flex>
              </TouchableArea>
            )}
          </Flex>
        </AnimatePresence>
      </Flex>
    </Flex>
  )
}
