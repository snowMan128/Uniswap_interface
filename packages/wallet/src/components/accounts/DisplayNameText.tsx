import { Flex, FlexProps, Text, TextProps } from 'ui/src'
import Unitag from 'ui/src/assets/graphics/unitag.svg'
import { iconSizes, IconSizeTokens } from 'ui/src/theme'
import { DisplayName, DisplayNameType } from 'wallet/src/features/wallet/types'

type DisplayNameProps = {
  displayName?: DisplayName
  unitagIconSize?: IconSizeTokens | number
  textProps?: TextProps
} & FlexProps

export function DisplayNameText({
  displayName,
  unitagIconSize = iconSizes.icon24,
  textProps,
  ...rest
}: DisplayNameProps): JSX.Element {
  const isUnitag = displayName?.type === DisplayNameType.Unitag

  return (
    <Flex centered row gap="$spacing4" {...rest}>
      <Text
        {...textProps}
        color={isUnitag ? '$accent1' : textProps?.color ?? '$neutral1'}
        flexShrink={1}
        numberOfLines={1}>
        {displayName?.name}
      </Text>
      {isUnitag ? (
        <Flex mt={-4}>
          <Unitag height={unitagIconSize} width={unitagIconSize} />
        </Flex>
      ) : null}
    </Flex>
  )
}
