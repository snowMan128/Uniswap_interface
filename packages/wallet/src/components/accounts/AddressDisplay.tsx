import { impactAsync } from 'expo-haptics'
import { PropsWithChildren, useMemo } from 'react'
import { FlexAlignType } from 'react-native'
import { ColorTokens, Flex, Icons, SpaceTokens, Text, TextProps, TouchableArea } from 'ui/src'
import { fonts } from 'ui/src/theme'
import { AccountIcon, AccountIconProps } from 'wallet/src/components/accounts/AccountIcon'
import { DisplayNameText } from 'wallet/src/components/accounts/DisplayNameText'
import { pushNotification } from 'wallet/src/features/notifications/slice'
import { AppNotificationType, CopyNotificationType } from 'wallet/src/features/notifications/types'
import { useAvatar, useDisplayName } from 'wallet/src/features/wallet/hooks'
import { DisplayNameType } from 'wallet/src/features/wallet/types'
import { useAppDispatch } from 'wallet/src/state'
import { ElementName } from 'wallet/src/telemetry/constants'
import { sanitizeAddressText, shortenAddress } from 'wallet/src/utils/addresses'
import { setClipboard } from 'wallet/src/utils/clipboard'

type AddressDisplayProps = {
  address: string
  allowFontScaling?: boolean
  hideAddressInSubtitle?: boolean
  size?: number
  variant?: keyof typeof fonts
  textColor?: ColorTokens
  captionTextColor?: ColorTokens
  captionVariant?: keyof typeof fonts
  direction?: 'row' | 'column'
  showCopy?: boolean
  showCopyWrapperButton?: boolean
  showAccountIcon?: boolean
  contentAlign?: FlexAlignType
  showIconBackground?: boolean
  showIconBorder?: boolean
  includeUnitagSuffix?: boolean
  textAlign?: FlexAlignType
  horizontalGap?: SpaceTokens
  notificationsBadgeContainer?: ({
    children,
    address,
  }: {
    children: JSX.Element
    address: string
  }) => JSX.Element
  gapBetweenLines?: SpaceTokens
} & Pick<AccountIconProps, 'showViewOnlyBadge'>

type CopyButtonWrapperProps = {
  onPress?: () => void
  backgroundColor?: string
}

function CopyButtonWrapper({
  children,
  onPress,
}: PropsWithChildren<CopyButtonWrapperProps>): JSX.Element {
  if (onPress) {
    return (
      <TouchableArea hapticFeedback hitSlop={16} testID={ElementName.Copy} onPress={onPress}>
        {children}
      </TouchableArea>
    )
  }

  return <>{children}</>
}

// This seems to work for most font sizes and screens, but could probably be improved and abstracted
// if we find more uses for it in other areas.
function getLineHeightForAdjustedFontSize(nameLength: number): number {
  // as name gets longer, number gets smaller down to 1, past 50 just 1
  const lineHeightBase = 50 - Math.min(49, nameLength)
  const scale = 1.2
  return lineHeightBase * scale
}

/** Helper component to display identicon and formatted address */

export function AddressDisplay({
  allowFontScaling = true,
  address,
  size = 24,
  variant = 'body1',
  textColor = '$neutral1',
  captionTextColor = '$neutral2',
  captionVariant = 'subheading2',
  hideAddressInSubtitle,
  direction = 'row',
  showCopy = false,
  showCopyWrapperButton = false,
  showAccountIcon = true,
  textAlign,
  contentAlign = 'center', // vertical alignment of all items
  showIconBackground,
  showIconBorder,
  horizontalGap = '$spacing12',
  showViewOnlyBadge = false,
  notificationsBadgeContainer,
  includeUnitagSuffix = false,
  gapBetweenLines = '$none',
}: AddressDisplayProps): JSX.Element {
  const dispatch = useAppDispatch()
  const displayName = useDisplayName(address, { includeUnitagSuffix })
  const { avatar } = useAvatar(address)

  const showAddressAsSubtitle =
    !hideAddressInSubtitle && displayName?.type !== DisplayNameType.Address

  const onPressCopyAddress = async (): Promise<void> => {
    if (!address) {
      return
    }
    await impactAsync()
    await setClipboard(address)
    dispatch(
      pushNotification({
        type: AppNotificationType.Copied,
        copyType: CopyNotificationType.Address,
      })
    )
  }

  // Extract sizes so copy icon can match font variants
  const mainSize = fonts[variant].fontSize
  const captionSize = fonts[captionVariant].fontSize
  const itemAlignment =
    textAlign || (!showAccountIcon || direction === 'column' ? 'center' : 'flex-start')

  const icon = useMemo(() => {
    return (
      <AccountIcon
        address={address}
        avatarUri={avatar}
        showBackground={showIconBackground}
        showBorder={showIconBorder}
        showViewOnlyBadge={showViewOnlyBadge}
        size={size}
      />
    )
  }, [address, avatar, showIconBackground, showIconBorder, showViewOnlyBadge, size])

  const name = displayName?.name ?? ''

  // since adjustsFontSizeToFit doesnt really work adjusting line height properly
  // manually adjust lineHeight things to keep vertical center
  const dynamicSizedTextVerticalStyles: TextProps =
    name.length > 20
      ? {
          adjustsFontSizeToFit: true,
          lineHeight: getLineHeightForAdjustedFontSize(name.length),
        }
      : {
          lineHeight: fonts[variant].lineHeight,
        }

  return (
    <Flex alignItems={contentAlign} flexDirection={direction} gap={horizontalGap}>
      {showAccountIcon &&
        (notificationsBadgeContainer
          ? notificationsBadgeContainer({ children: icon, address })
          : icon)}
      <Flex shrink alignItems={itemAlignment} gap={gapBetweenLines}>
        <CopyButtonWrapper
          onPress={showCopy && !showAddressAsSubtitle ? onPressCopyAddress : undefined}>
          <Flex centered row gap="$spacing12">
            <DisplayNameText
              displayName={displayName}
              textProps={{
                adjustsFontSizeToFit: true,
                allowFontScaling,
                color: textColor,
                ellipsizeMode: 'tail',
                fontFamily: '$heading',
                fontSize: mainSize,
                numberOfLines: 1,
                testID: `address-display/name/${displayName?.name}`,
                ...dynamicSizedTextVerticalStyles,
              }}
              unitagIconSize={mainSize}
            />
            {showCopy && !showAddressAsSubtitle && (
              <Icons.CopySheets color="$neutral1" size={mainSize} />
            )}
          </Flex>
        </CopyButtonWrapper>
        {showAddressAsSubtitle && (
          <AddressSubtitle
            address={address}
            captionSize={captionSize}
            captionTextColor={captionTextColor}
            captionVariant={captionVariant}
            showCopy={showCopy}
            showCopyWrapperButton={showCopyWrapperButton}
            onPressCopyAddress={onPressCopyAddress}
          />
        )}
      </Flex>
    </Flex>
  )
}

const AddressSubtitle = ({
  address,
  captionTextColor,
  captionVariant,
  captionSize,
  showCopy,
  showCopyWrapperButton,
  onPressCopyAddress,
}: { captionSize: number; onPressCopyAddress: () => Promise<void> } & Pick<
  AddressDisplayProps,
  'address' | 'captionTextColor' | 'captionVariant' | 'showCopy' | 'showCopyWrapperButton'
>): JSX.Element => (
  <CopyButtonWrapper onPress={showCopy ? onPressCopyAddress : undefined}>
    <Flex
      centered
      row
      backgroundColor={showCopyWrapperButton ? '$DEP_backgroundOverlay' : '$transparent'}
      borderRadius="$roundedFull"
      gap="$spacing4"
      mt={showCopyWrapperButton ? '$spacing8' : '$none'}
      px={showCopyWrapperButton ? '$spacing8' : '$none'}
      py={showCopyWrapperButton ? '$spacing4' : '$none'}>
      <Text color={captionTextColor} variant={captionVariant}>
        {sanitizeAddressText(shortenAddress(address))}
      </Text>
      {showCopy && <Icons.CopySheets color={captionTextColor} size={captionSize} />}
    </Flex>
  </CopyButtonWrapper>
)
