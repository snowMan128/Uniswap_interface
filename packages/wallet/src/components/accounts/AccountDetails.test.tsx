import { AccountDetails } from 'wallet/src/components/accounts/AccountDetails'
import { account } from 'wallet/src/test/fixtures'
import { render } from 'wallet/src/test/test-utils'

describe(AccountDetails, () => {
  it('renders without error', () => {
    const tree = render(<AccountDetails address={account.address} iconSize={50} />)

    expect(tree.toJSON()).toMatchSnapshot()
  })

  it('renders without error with chevron', () => {
    const tree = render(<AccountDetails chevron address={account.address} iconSize={50} />)

    expect(tree.toJSON()).toMatchSnapshot()
  })
})
