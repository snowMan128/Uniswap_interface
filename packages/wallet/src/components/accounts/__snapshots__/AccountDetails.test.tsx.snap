// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountDetails renders without error 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "flexShrink": 1,
      "gap": 16,
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      {
        "alignItems": "stretch",
        "flex": 1,
        "flexDirection": "row",
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "stretch",
            "backgroundColor": "transparent",
            "borderBottomColor": "transparent",
            "borderBottomLeftRadius": 999999,
            "borderBottomRightRadius": 999999,
            "borderBottomWidth": 0,
            "borderLeftColor": "transparent",
            "borderLeftWidth": 0,
            "borderRightColor": "transparent",
            "borderRightWidth": 0,
            "borderStyle": "solid",
            "borderTopColor": "transparent",
            "borderTopLeftRadius": 999999,
            "borderTopRightRadius": 999999,
            "borderTopWidth": 0,
            "flexDirection": "column",
            "position": "relative",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderBottomLeftRadius": 999999,
              "borderBottomRightRadius": 999999,
              "borderTopLeftRadius": 999999,
              "borderTopRightRadius": 999999,
              "flexDirection": "column",
              "height": 50,
              "justifyContent": "center",
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "width": 50,
            }
          }
        >
          <div
            style={
              {
                "height": 50,
                "position": "relative",
                "width": 50,
              }
            }
          >
            <div
              style={
                {
                  "height": 50,
                  "overflow": "visible",
                  "position": "absolute",
                  "width": 50,
                }
              }
            >
              <svg
                viewBox="0 0 50 50"
              >
                <defs>
                  <defs>
                    <mask
                      id="container-mask0x82D56A352367453f74FC0dC7B071b311da373Fa650"
                    >
                      <rect
                        fill="white"
                        height="100%"
                        width="100%"
                        x="0"
                        y="0"
                      />
                      <g
                        transform="scale(1.3888888888888888) 
 translate(0, 0)"
                      >
                        <path
                          clipRule="evenodd"
                          d="M31 33V16C31 8.8203 25.1797 3 18 3C10.8203 3 5 8.8203 5 16V33H31ZM18 0C9.16344 0 2 7.16344 2 16V35.807C2 35.9136 2.08642 36 2.19303 36H33.807C33.9136 36 34 35.9136 34 35.807V16C34 7.16344 26.8366 0 18 0Z"
                          fill="black"
                          fillRule="evenodd"
                        />
                      </g>
                    </mask>
                    <mask
                      id="shape-mask0x82D56A352367453f74FC0dC7B071b311da373Fa650"
                    >
                      <rect
                        fill="white"
                        height="100%"
                        width="100%"
                        x="0"
                        y="0"
                      />
                      <g
                        transform="scale(1.3888888888888888) 
 translate(10, 10)"
                      >
                        <path
                          d="M8 8C8 8 13.1678 6.70089 14.2197 5.649C15.2716 4.59712 15.2583 2.87836 14.19 1.81004C13.1216 0.741719 11.4029 0.728395 10.351 1.78028C9.29911 2.83216 8 8 8 8Z"
                          fill="black"
                        />
                        <path
                          d="M8 8C8 8 13.1678 9.29911 14.2197 10.351C15.2716 11.4029 15.2583 13.1216 14.19 14.19C13.1216 15.2583 11.4029 15.2716 10.351 14.2197C9.29911 13.1678 8 8 8 8Z"
                          fill="black"
                        />
                        <path
                          d="M1.78028 10.351C2.83216 9.29911 8 8 8 8C8 8 6.70089 13.1678 5.649 14.2197C4.59712 15.2716 2.87836 15.2583 1.81004 14.19C0.741719 13.1216 0.728395 11.4029 1.78028 10.351Z"
                          fill="black"
                        />
                        <path
                          d="M8 8C8 8 2.83216 6.70089 1.78028 5.649C0.728395 4.59712 0.741719 2.87836 1.81004 1.81004C2.87836 0.741719 4.59712 0.728395 5.649 1.78028C6.70089 2.83216 8 8 8 8Z"
                          fill="black"
                        />
                      </g>
                    </mask>
                    <mask
                      id="mask0x82D56A352367453f74FC0dC7B071b311da373Fa650"
                    >
                      <g
                        fill="white"
                      >
                        <g
                          mask="url(#shape-mask0x82D56A352367453f74FC0dC7B071b311da373Fa650)"
                        >
                          <g
                            transform="scale(1.3888888888888888)"
                          >
                            <path
                              clipRule="evenodd"
                              d="M31 33V16C31 8.8203 25.1797 3 18 3C10.8203 3 5 8.8203 5 16V33H31ZM18 0C9.16344 0 2 7.16344 2 16V35.807C2 35.9136 2.08642 36 2.19303 36H33.807C33.9136 36 34 35.9136 34 35.807V16C34 7.16344 26.8366 0 18 0Z"
                              fillRule="evenodd"
                            />
                          </g>
                        </g>
                        <g
                          mask="url(#container-mask0x82D56A352367453f74FC0dC7B071b311da373Fa650)"
                        >
                          <g
                            transform="scale(1.3888888888888888)
        translate(10, 10)"
                          >
                            <path
                              d="M8 8C8 8 13.1678 6.70089 14.2197 5.649C15.2716 4.59712 15.2583 2.87836 14.19 1.81004C13.1216 0.741719 11.4029 0.728395 10.351 1.78028C9.29911 2.83216 8 8 8 8Z"
                            />
                            <path
                              d="M8 8C8 8 13.1678 9.29911 14.2197 10.351C15.2716 11.4029 15.2583 13.1216 14.19 14.19C13.1216 15.2583 11.4029 15.2716 10.351 14.2197C9.29911 13.1678 8 8 8 8Z"
                            />
                            <path
                              d="M1.78028 10.351C2.83216 9.29911 8 8 8 8C8 8 6.70089 13.1678 5.649 14.2197C4.59712 15.2716 2.87836 15.2583 1.81004 14.19C0.741719 13.1216 0.728395 11.4029 1.78028 10.351Z"
                            />
                            <path
                              d="M8 8C8 8 2.83216 6.70089 1.78028 5.649C0.728395 4.59712 0.741719 2.87836 1.81004 1.81004C2.87836 0.741719 4.59712 0.728395 5.649 1.78028C6.70089 2.83216 8 8 8 8Z"
                            />
                          </g>
                        </g>
                      </g>
                    </mask>
                  </defs>
                  <linearGradient
                    id="gradient0x82D56A352367453f74FC0dC7B071b311da373Fa650"
                  >
                    <stop
                      offset="0%"
                      stopColor="#6100FF"
                    />
                    <stop
                      offset="100%"
                      stopColor="#76D191"
                    />
                  </linearGradient>
                  <filter
                    height="200%"
                    id="blur0x82D56A352367453f74FC0dC7B071b311da373Fa650"
                    width="200%"
                    x="-50%"
                    y="-50%"
                  >
                    <feGaussianBlur
                      in="SourceGraphic"
                      stdDeviation={16.666666666666668}
                    />
                  </filter>
                </defs>
                <g
                  mask="url(#mask0x82D56A352367453f74FC0dC7B071b311da373Fa650)"
                >
                  <rect
                    fill="url(#gradient0x82D56A352367453f74FC0dC7B071b311da373Fa650)"
                    height="100%"
                    width="100%"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="black"
                    height="100%"
                    opacity={0.08}
                    width="100%"
                    x="0"
                    y="0"
                  />
                  <ellipse
                    cx={25}
                    cy={0}
                    fill="#D3EBA3"
                    filter="url(#blur0x82D56A352367453f74FC0dC7B071b311da373Fa650)"
                    rx={25}
                    ry={25}
                  />
                </g>
              </svg>
            </div>
          </div>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "flex-start",
            "flexDirection": "column",
            "flexShrink": 1,
            "gap": 0,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 12,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "gap": 4,
                "justifyContent": "center",
              }
            }
          >
            <Text
              adjustsFontSizeToFit={true}
              allowFontScaling={true}
              maxFontSizeMultiplier={1.4}
              numberOfLines={1}
              style={
                {
                  "color": "#222222",
                  "flexShrink": 1,
                  "fontFamily": "Basel-Book",
                  "fontSize": 15,
                  "lineHeight": 20,
                }
              }
              suppressHighlighting={true}
              testID="address-display/name/0x​82D5...3Fa6"
            >
              0x​82D5...3Fa6
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "flexShrink": 1,
        "gap": 4,
        "justifyContent": "flex-end",
      }
    }
  >
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.4}
      style={
        {
          "color": "#7D7D7D",
          "fontFamily": "Basel-Book",
          "fontSize": 15,
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      0x82D5...3Fa6
    </Text>
  </View>
</View>
`;

exports[`AccountDetails renders without error with chevron 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "flexShrink": 1,
      "gap": 16,
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      {
        "alignItems": "stretch",
        "flex": 1,
        "flexDirection": "row",
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "stretch",
            "backgroundColor": "transparent",
            "borderBottomColor": "transparent",
            "borderBottomLeftRadius": 999999,
            "borderBottomRightRadius": 999999,
            "borderBottomWidth": 0,
            "borderLeftColor": "transparent",
            "borderLeftWidth": 0,
            "borderRightColor": "transparent",
            "borderRightWidth": 0,
            "borderStyle": "solid",
            "borderTopColor": "transparent",
            "borderTopLeftRadius": 999999,
            "borderTopRightRadius": 999999,
            "borderTopWidth": 0,
            "flexDirection": "column",
            "position": "relative",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "borderBottomLeftRadius": 999999,
              "borderBottomRightRadius": 999999,
              "borderTopLeftRadius": 999999,
              "borderTopRightRadius": 999999,
              "flexDirection": "column",
              "height": 50,
              "justifyContent": "center",
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
              "width": 50,
            }
          }
        >
          <div
            style={
              {
                "height": 50,
                "position": "relative",
                "width": 50,
              }
            }
          >
            <div
              style={
                {
                  "height": 50,
                  "overflow": "visible",
                  "position": "absolute",
                  "width": 50,
                }
              }
            >
              <svg
                viewBox="0 0 50 50"
              >
                <defs>
                  <defs>
                    <mask
                      id="container-mask0x82D56A352367453f74FC0dC7B071b311da373Fa650"
                    >
                      <rect
                        fill="white"
                        height="100%"
                        width="100%"
                        x="0"
                        y="0"
                      />
                      <g
                        transform="scale(1.3888888888888888) 
 translate(0, 0)"
                      >
                        <path
                          clipRule="evenodd"
                          d="M31 33V16C31 8.8203 25.1797 3 18 3C10.8203 3 5 8.8203 5 16V33H31ZM18 0C9.16344 0 2 7.16344 2 16V35.807C2 35.9136 2.08642 36 2.19303 36H33.807C33.9136 36 34 35.9136 34 35.807V16C34 7.16344 26.8366 0 18 0Z"
                          fill="black"
                          fillRule="evenodd"
                        />
                      </g>
                    </mask>
                    <mask
                      id="shape-mask0x82D56A352367453f74FC0dC7B071b311da373Fa650"
                    >
                      <rect
                        fill="white"
                        height="100%"
                        width="100%"
                        x="0"
                        y="0"
                      />
                      <g
                        transform="scale(1.3888888888888888) 
 translate(10, 10)"
                      >
                        <path
                          d="M8 8C8 8 13.1678 6.70089 14.2197 5.649C15.2716 4.59712 15.2583 2.87836 14.19 1.81004C13.1216 0.741719 11.4029 0.728395 10.351 1.78028C9.29911 2.83216 8 8 8 8Z"
                          fill="black"
                        />
                        <path
                          d="M8 8C8 8 13.1678 9.29911 14.2197 10.351C15.2716 11.4029 15.2583 13.1216 14.19 14.19C13.1216 15.2583 11.4029 15.2716 10.351 14.2197C9.29911 13.1678 8 8 8 8Z"
                          fill="black"
                        />
                        <path
                          d="M1.78028 10.351C2.83216 9.29911 8 8 8 8C8 8 6.70089 13.1678 5.649 14.2197C4.59712 15.2716 2.87836 15.2583 1.81004 14.19C0.741719 13.1216 0.728395 11.4029 1.78028 10.351Z"
                          fill="black"
                        />
                        <path
                          d="M8 8C8 8 2.83216 6.70089 1.78028 5.649C0.728395 4.59712 0.741719 2.87836 1.81004 1.81004C2.87836 0.741719 4.59712 0.728395 5.649 1.78028C6.70089 2.83216 8 8 8 8Z"
                          fill="black"
                        />
                      </g>
                    </mask>
                    <mask
                      id="mask0x82D56A352367453f74FC0dC7B071b311da373Fa650"
                    >
                      <g
                        fill="white"
                      >
                        <g
                          mask="url(#shape-mask0x82D56A352367453f74FC0dC7B071b311da373Fa650)"
                        >
                          <g
                            transform="scale(1.3888888888888888)"
                          >
                            <path
                              clipRule="evenodd"
                              d="M31 33V16C31 8.8203 25.1797 3 18 3C10.8203 3 5 8.8203 5 16V33H31ZM18 0C9.16344 0 2 7.16344 2 16V35.807C2 35.9136 2.08642 36 2.19303 36H33.807C33.9136 36 34 35.9136 34 35.807V16C34 7.16344 26.8366 0 18 0Z"
                              fillRule="evenodd"
                            />
                          </g>
                        </g>
                        <g
                          mask="url(#container-mask0x82D56A352367453f74FC0dC7B071b311da373Fa650)"
                        >
                          <g
                            transform="scale(1.3888888888888888)
        translate(10, 10)"
                          >
                            <path
                              d="M8 8C8 8 13.1678 6.70089 14.2197 5.649C15.2716 4.59712 15.2583 2.87836 14.19 1.81004C13.1216 0.741719 11.4029 0.728395 10.351 1.78028C9.29911 2.83216 8 8 8 8Z"
                            />
                            <path
                              d="M8 8C8 8 13.1678 9.29911 14.2197 10.351C15.2716 11.4029 15.2583 13.1216 14.19 14.19C13.1216 15.2583 11.4029 15.2716 10.351 14.2197C9.29911 13.1678 8 8 8 8Z"
                            />
                            <path
                              d="M1.78028 10.351C2.83216 9.29911 8 8 8 8C8 8 6.70089 13.1678 5.649 14.2197C4.59712 15.2716 2.87836 15.2583 1.81004 14.19C0.741719 13.1216 0.728395 11.4029 1.78028 10.351Z"
                            />
                            <path
                              d="M8 8C8 8 2.83216 6.70089 1.78028 5.649C0.728395 4.59712 0.741719 2.87836 1.81004 1.81004C2.87836 0.741719 4.59712 0.728395 5.649 1.78028C6.70089 2.83216 8 8 8 8Z"
                            />
                          </g>
                        </g>
                      </g>
                    </mask>
                  </defs>
                  <linearGradient
                    id="gradient0x82D56A352367453f74FC0dC7B071b311da373Fa650"
                  >
                    <stop
                      offset="0%"
                      stopColor="#6100FF"
                    />
                    <stop
                      offset="100%"
                      stopColor="#76D191"
                    />
                  </linearGradient>
                  <filter
                    height="200%"
                    id="blur0x82D56A352367453f74FC0dC7B071b311da373Fa650"
                    width="200%"
                    x="-50%"
                    y="-50%"
                  >
                    <feGaussianBlur
                      in="SourceGraphic"
                      stdDeviation={16.666666666666668}
                    />
                  </filter>
                </defs>
                <g
                  mask="url(#mask0x82D56A352367453f74FC0dC7B071b311da373Fa650)"
                >
                  <rect
                    fill="url(#gradient0x82D56A352367453f74FC0dC7B071b311da373Fa650)"
                    height="100%"
                    width="100%"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="black"
                    height="100%"
                    opacity={0.08}
                    width="100%"
                    x="0"
                    y="0"
                  />
                  <ellipse
                    cx={25}
                    cy={0}
                    fill="#D3EBA3"
                    filter="url(#blur0x82D56A352367453f74FC0dC7B071b311da373Fa650)"
                    rx={25}
                    ry={25}
                  />
                </g>
              </svg>
            </div>
          </div>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "flex-start",
            "flexDirection": "column",
            "flexShrink": 1,
            "gap": 0,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "gap": 12,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "gap": 4,
                "justifyContent": "center",
              }
            }
          >
            <Text
              adjustsFontSizeToFit={true}
              allowFontScaling={true}
              maxFontSizeMultiplier={1.4}
              numberOfLines={1}
              style={
                {
                  "color": "#222222",
                  "flexShrink": 1,
                  "fontFamily": "Basel-Book",
                  "fontSize": 15,
                  "lineHeight": 20,
                }
              }
              suppressHighlighting={true}
              testID="address-display/name/0x​82D5...3Fa6"
            >
              0x​82D5...3Fa6
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "flexShrink": 1,
        "gap": 4,
        "justifyContent": "flex-end",
      }
    }
  >
    <Text
      allowFontScaling={true}
      maxFontSizeMultiplier={1.4}
      style={
        {
          "color": "#7D7D7D",
          "fontFamily": "Basel-Book",
          "fontSize": 15,
          "lineHeight": 20,
        }
      }
      suppressHighlighting={true}
    >
      0x82D5...3Fa6
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "borderBottomLeftRadius": 999999,
          "borderBottomRightRadius": 999999,
          "borderTopLeftRadius": 999999,
          "borderTopRightRadius": 999999,
          "flexDirection": "column",
          "justifyContent": "center",
          "transform": [
            {
              "rotate": "180deg",
            },
          ],
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="50"
        bbWidth="50"
        fill="none"
        focusable={false}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="currentColor"
        strokeWidth={8}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "color": "#7D7D7D",
              "height": 50,
              "width": 50,
            },
            {
              "flex": 0,
              "height": 50,
              "width": 50,
            },
          ]
        }
        tintColor="#7D7D7D"
        vbHeight={24}
        vbWidth={24}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
            ]
          }
          stroke={
            {
              "type": 2,
            }
          }
          strokeWidth="8"
        >
          <RNSVGPath
            d="M15 6L9 12L15 18"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="3"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
